File: app/src/main/java/io/oldering/tvfoot/red/di/module/MatchesActivityModule.java
package io.oldering.tvfoot.red.di.module;

import android.app.Activity;
import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import io.oldering.tvfoot.red.di.component.MatchesActivitySubComponent;
import io.oldering.tvfoot.red.matches.MatchesActivity;

@Module(subcomponents = MatchesActivitySubComponent.class) //
public abstract class MatchesActivityModule {
  @Binds @IntoMap @ActivityKey(MatchesActivity.class) //
  abstract AndroidInjector.Factory<? extends Activity> bindMatchesActivityInjectorFactory(
      MatchesActivitySubComponent.Builder builder);
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/MatchesActivityModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import android.app.Activity
import com.benoitquenaudon.tvfoot.red.app.domain.matches.MatchesActivity
import com.benoitquenaudon.tvfoot.red.app.domain.matches.filters.FiltersFragment
import com.benoitquenaudon.tvfoot.red.injection.scope.FragmentScope
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector


@Module
abstract class MatchesActivityModule {
  @Binds abstract fun provideActivity(activity: MatchesActivity): Activity

  @FragmentScope
  @ContributesAndroidInjector
  internal abstract fun filtersFragment(): FiltersFragment
}